src/wasm-lib/kcl/src/errors.rs

```diff
@@ -147,6 +147,157 @@ impl KclErrorWithOutputs {
             filenames: Default::default(),
         }
     }
+    pub async fn into_miette_report_with_outputs(
+        self,
+        fs: std::sync::Arc<crate::fs::FileManager>,
+    ) -> anyhow::Result<ReportWithOutputs> {
+        let mut source_ranges = self.error.source_ranges();
+
+        // Pop off the first source range to get the filename.
+        let first_source_range = source_ranges
+            .pop()
+            .ok_or_else(|| anyhow::anyhow!("No source ranges found"))?;
+
+        let filename = self
+            .filenames
+            .get(&first_source_range.module_id())
+            .map(|p| p.to_string())
+            .ok_or_else(|| {
+                anyhow::anyhow!(
+                    "Could not find filename for module id: {:?}",
+                    first_source_range.module_id()
+                )
+            })?;
+        let kcl_source = std::str::from_utf8(&fs.read(&filename, Default::default()).await?)?.to_string();
+
+        let mut related = Vec::new();
+        for source_range in source_ranges {
+            let module_id = source_range.module_id();
+            let filename = self
+                .filenames
+                .get(&module_id)
+                .map(|p| p.to_string())
+                .ok_or_else(|| anyhow::anyhow!("Could not find filename for module id: {:?}", module_id))?;
+            let error = self.error.override_source_ranges(vec![source_range]);
+            let report = Report {
+                error,
+                kcl_source: std::str::from_utf8(&fs.read(&filename, Default::default()).await?)?.to_string(),
+                filename,
+            };
+            related.push(report);
+        }
+
+        Ok(ReportWithOutputs {
+            error: self,
+            kcl_source,
+            filename,
+            related,
+        })
+    }
+}
+
+impl IntoDiagnostic for KclErrorWithOutputs {
+    async fn to_lsp_diagnostics(&self, code: &str) -> Vec<Diagnostic> {
```

Comment:
Could we reuse the KclError impl rather than write a whole new one?

Reply:
well KclError does not have the file data thats why i needed this in the first place KclErrorWithOutputs is the only thing that has a map of files to module numbers

Reply:
they are two different things and basically KclError is kinda useless for anything assembly related since it lacks the data hence in all upstreams and tests now i use the KclErrorWithOutputs

Reply:
Could we remove the KclError impl then?

Reply:
i think the ui still agressively uses it, i dont know tho, kinda seems like it could be another PR

Reply:
id love to get rid of it but like they are passed literally everywhjere AND then at the end returned with the outputs it would be like 300 places to change

Reply:
anywways this is a lot cleaner now cant believe i was reading the file everytime like a doofus